comp/metadata/inventorychecks/inventorychecksimpl/inventorychecks.go

```diff
@@ -270,3 +294,9 @@ func (ic *inventorychecksImpl) writePayloadAsJSON(w http.ResponseWriter, _ *http
 	}
 	w.Write(scrubbed)
 }
+
+func init() {
+	expvar.Publish("inventories", expvar.Func(func() interface{} {
+		return expvarCallback()
+	}))
```

Comment:
Question: 

What would happen if we move 
```go
expvar.Publish("inventories", expvar.Func(func() interface{} {
    return ic.getPayload()
})
```

Inside the Start hook?

Would it work? Could it help us remove the global and the init function?

Reply:
That what I did at the beginning, but it was panicking during the unit tests due to multiple `expvar.Publish` call with the same "key".
It's very unlikely that could happen in real context (fx only instantiate one time `inventorychecks`) but we never know ?
I thought about using testSuite to only rely on one instance of `inventorychecks`, but choose the other solution (also because it's supposed to be a temporary fix until the migration), without strong opinion on it.

Reply:
> That what I did at the beginning, but it was panicking during the unit tests due to multiple expvar.Publish call with the same "key".

I mean feels weird to create code around the tests, it should be the other way around. 

I'm not sure how difficult or time consuming to modify the tests o only one instance of `inventorychecks` is created, nor if there would be other broken tests, but I would try to modify the tests and if it feasible I would change the implementation to include the expvar function registration in the start hook.

If it turns out the changes are to many, we can leave the solution as it is. As you mentioned it is temporary (but for how long 😉)

Reply:
You're totally right ! I changed the code based on your comment and it's better, smarter and shorter !